lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -171,31 +174,31 @@ def relationships_for(serializer)
           end
         end
 
-        def included_resources(include_tree)
+        def included_resources(include_tree, primary_data)
           included = []
 
           serializer.associations(include_tree).each do |association|
-            add_included_resources_for(association.serializer, include_tree[association.key], included)
+            add_included_resources_for(association.serializer, include_tree[association.key], primary_data, included)
           end
 
           included
         end
 
-        def add_included_resources_for(serializer, include_tree, included)
+        def add_included_resources_for(serializer, include_tree, primary_data, included)
           if serializer.respond_to?(:each)
-            serializer.each { |s| add_included_resources_for(s, include_tree, included) }
+            serializer.each { |s| add_included_resources_for(s, include_tree, primary_data, included) }
```

Comment:
and primary_data (which is `[primary_data]`) is a new argument to `add_included_resources_for`